b+tree.ts

```diff
@@ -1765,11 +1781,26 @@ class BNodeInternal<K,V> extends BNode<K,V> {
     return false;
   }
 
+  /**
+   * Move children from `rhs` into this.
+   * `rhs` must be part of this tree, and be removed from it after this call
+   * (otherwise isShared for its children could be incorrect).
+   */
   mergeSibling(rhs: BNode<K,V>, maxNodeSize: number) {
     // assert !this.isShared;
     var oldLength = this.keys.length;
     this.keys.push.apply(this.keys, rhs.keys);
-    this.children.push.apply(this.children, (rhs as any as BNodeInternal<K,V>).children);
+    const rhsChildren = (rhs as any as BNodeInternal<K,V>).children;
+    this.children.push.apply(this.children, rhsChildren);
+
+    if (rhs.isShared) {
```

Comment:
Is it an improvement to check `(rhs.isShared && !this.isShared)` here?

Reply:
If `this.isShared`, then we shouldn't have modified `this.children`. The caller must ensure `!this.isShared` before calling this method.

Reply:
@titrindl is right that such a check is unneeded: this is never shared at this point, and it being shared would be illegal as its modified in this function.

Additionally, these children need to be marked as shared because a parent is being added and the old parent is not being removed. This is true regardless of this being shared. The only reason we can skip marking these when rhs is not shared is because we know that in that case the old parent is being removed (we require that the caller does this): the shared status of `this` has no impact on how many parents the children will have after this function completes.